data_rentgen/db/scripts/clean_partitions.py

```diff
@@ -0,0 +1,203 @@
+        if month:
+            month = int(month)
+            granularity = "month"
+        else:
+            month = 1
+        if day:
+            day = int(day)
+            granularity = "day"
+        else:
+            day = 1
+        tables[name].partitions.append(date(year, month, day))
+        tables[name].granularity = granularity
+    return tables
+
+
+def get_partitions(tables: dict[str, Partition], end_date: date):
+    for table_name, value in tables.items():
+        partitions_to_remove = [partition for partition in value.partitions if partition < end_date]
+        tables[table_name].partitions = partitions_to_remove
+    return tables
+
+
+def get_query(tables: dict[str, Partition], query_type: str):
+    queries = []
+    query_template = ""
+    match query_type:
+        case "detach":
+            query_template = "ALTER TABLE {table_name} DETACH PARTITION {table_name}_{partition};"
+        case "remove":
+            query_template = "DROP TABLE {table_name}_{partition};"
+        case "truncate":
+            query_template = "TRUNCATE TABLE {table_name}_{partition};"
+    for table_name, value in tables.items():
+        queries_per_table = [
+            query_template.format(
+                table_name=table_name,
+                partition=partition.strftime(granularity_output_format[value.granularity]),
+            )
+            for partition in value.partitions
+        ]
+        queries.extend(queries_per_table)
+    return queries
+
+
+def show_removing_partitions(tables: dict[str, Partition]):
+    for table_name, value in tables.items():
+        logger.info("Removing next partitions for table: %s", table_name)
+        partitions_names = " ".join(
+            [
+                f"{table_name}_{partition.strftime(granularity_output_format[value.granularity])}"
+                for partition in value.partitions
+            ],
+        )
+        logger.info(partitions_names)
+
+
+async def detach_partitions(tables: dict[str, Partition], session: AsyncSession):
+    partitions_to_detach = get_query(tables, "detach")
+    for query in partitions_to_detach:
+        logger.info("Detach partitions with query: %s", query)
+        await session.execute(text(query))
+        await session.commit()
+
+
+async def remove_data(tables: dict[str, Partition], session: AsyncSession):
+    partitions_to_remove = get_query(tables, "remove")
+    for query in partitions_to_remove:
+        logger.info("Remove table with query: %s", query)
+        await session.execute(text(query))
+        await session.commit()
+
+
+async def truncate_data(tables: dict[str, Partition], session: AsyncSession):
+    partitions_to_truncate = get_query(tables, "truncate")
+    for query in partitions_to_truncate:
+        logger.info("Truncate partition with query: %s", query)
+        await session.execute(text(query))
+        await session.commit()
+
+
+async def main(args: list[str]) -> None:
+    setup_logging(LoggingSettings())
```

Comment:
Let's move this to `__main__`, as tests for this script may change pytest logging level and format